en/mavgen_c/README.md

```diff
@@ -59,16 +60,164 @@ When compiling the project, we recommend that you specify the top-level output d
 $ gcc ... -I generated/include -I generated/include/common ...
 ```
 
-## Multiple Streams ("channels") {#streams}
+## Multiple Streams ("channels") {#channels}
 
-The C MAVLink library utilizes a "channel" metaphor to allow for simultaneous processing of multiple, independent MAVLink streams in the same program. 
-It is therefore important to use the correct channel for each operation as all receiving and transmitting functions provided by MAVLink require a channel.
+The C MAVLink library utilizes a "channel" metaphor to allow for simultaneous processing of multiple, independent MAVLink streams in the same program.
+All receiving and transmitting functions provided by this library require a channel, and it is important to use the correct channel for each operation.
 
+Up to 4 channels can be defined on MCU/ARM7 while up to 16 may be defined on Linux/Windows.
 If only one MAVLink stream exists, channel 0 should be used by specifying the `MAVLINK_COMM_0` constant.
 
+
 ## Receiving
 
-MAVLink reception is then done using `mavlink_helpers.h:mavlink_parse_char()`.
+MAVLink reception/decoding is done in a number of phases: 
+1. Parse the incoming stream into objects representing each packet (`mavlink_message_t`).
+1. Decode the MAVLink message contained in the packet payload into a C struct (that has fields mapping the original XML definition).
+
+These steps are demonstrated below.
+
+### Parsing Packets
+
+The `mavlink_parse_char()` convenience function ([mavlink_helpers.h](https://github.com/mavlink/c_library_v2/blob/master/mavlink_helpers.h)) is used to parse incoming MAVLink data.
```

Comment:
elipsis to indicate this isn't a void-taking function...

Reply:
Hmmm. Won't most people just assume that `mavlink_parse_char()` is a function name? Generally I add the elipsis if I am including the return type - in that case I am showing enough that people might thing that it is a signature.

Reply:
Not when you put the parentheses in.  Then it starts to look like a signature.  Could could leave the prens off.

Reply:
I have left this as-is for now:
- The parenthesis make it immediately obvious that you're talking about a function.
- I personally never assume I'm looking at a signature unless I see a return type or it says that it is a signature.
- This pattern is used throughout the doc set.

So this should be changed globally if there is general agreement it is a problem. But for now, I don't see it as a big one.